### ISSUE TITLE
mouth

### ISSUE THREAD

**Issue by zhulei1228** [MarekKowalski/FaceSwap#21]
Can the mouth not be swapped?

**Comment by MarekKowalski** [comment]
It can be in principle, you would need to add triangles that cover the mouth to the mesh. The issue is that it would be difficult to make the swapped mouth look realistic. 

Imagine a situation where the photo you are taking the texture from has the mouth closed. Then imagine the person in front of the camera opens their mouth. The effect would not be very satisfying as the texture for the whole open mouth would have to be generated from a very thin piece of the source image.

There are ways of dealing with this through machine learning, you can see [this paper](http://www.hao-li.com/publications/papers/siggraphAsia2018PAGAN.pdf) as an example.

**Comment by zhulei1228** [comment]
> It can be in principle, you would need to add triangles that cover the mouth to the mesh. The issue is that it would be difficult to make the swapped mouth look realistic.
> 
> Imagine a situation where the photo you are taking the texture from has the mouth closed. Then imagine the person in front of the camera opens their mouth. The effect would not be very satisfying as the texture for the whole open mouth would have to be generated from a very thin piece of the source image.
> 
> There are ways of dealing with this through machine learning, you can see [this paper](http://www.hao-li.com/publications/papers/siggraphAsia2018PAGAN.pdf) as an example.

ok,thank you.But the eyes always don't blink.And when a frame has no face,the dst face still appear

**Comment by MarekKowalski** [comment]
Yes, both of those are true. If you didn't want the eyes to appear you could remove the triangles that correspond to the eyes from the mesh. I think that the eyes are a bit less jarring than the mouth as eyelid movement changes the shape of the face much more.

**Comment by zhulei1228** [comment]
> Yes, both of those are true. If you didn't want the eyes to appear you could remove the triangles that correspond to the eyes from the mesh. I think that the eyes are a bit less jarring than the mouth as eyelid movement changes the shape of the face much more.

How to change blendshapes to change expressions?

**Comment by MarekKowalski** [comment]
Could you elaborate on what you mean by that?
